sources/rest_api/paginators.py

```diff
@@ -133,19 +134,46 @@ class JSONResponsePaginator(BaseNextUrlPaginator):
 
     def __init__(
         self,
-        next_key: Union[str, Sequence[str]] = "next",
+        next_url_path: jsonpath.TJsonPath = "next",
     ):
         """
         Args:
-            next_key (str, optional): The key in the JSON response that
-                contains the next page URL. Defaults to 'next'.
+            next_url_path: The JSON path to the key that contains the next page URL in the response.
+                Defaults to 'next'.
         """
         super().__init__()
-        self.next_key = next_key
-        self._next_key_accessor = create_nested_accessor(next_key)
+        self.next_url_path = jsonpath.compile_path(next_url_path)
 
     def update_state(self, response: Response) -> None:
-        try:
-            self.next_reference = self._next_key_accessor(response.json())
-        except KeyError:
-            self.next_reference = None
+        values = jsonpath.find_values(self.next_url_path, response.json())
+        self._next_reference = values[0] if values else None
+
+
+class JSONResponseCursorPaginator(BasePaginator):
```

Comment:
cool. but this one we won't be able to detect from data :/ we need to implement some simple heuristics in openAPI repo that use request and response entities to do that.
@steinitzu I'll ping you there

Reply:
True, we can't detect the query param. But we can detect it from openapi for sure, or configured manually.